vscode/src/programConfig.ts

```diff
@@ -76,10 +75,6 @@ export function getActiveQSharpDocumentUri(): vscode.Uri | undefined {
 export async function getProgramForDocument(
   docUri: vscode.Uri,
 ): Promise<FullProgramConfigOrError> {
-  if (Utils.extname(docUri) !== ".qs") {
-    return { success: false, errorMsg: `${docUri.fsPath} is not a Q# file` };
```

Comment:
A lot of other paths call getActiveProgram (esp. commands like RE and circuits), which ends with `return await getProgramForDocument(docUri);`, and check `success` on the result to see if they should error or not. Did you check those code paths to make sure this change is OK?

Reply:
(they seem OK at first glance to me - just double checking)

Reply:
If you look farther up in getActiveProgram, it checks the language setting for the document. I'm of the opinion that the setting should supersede extension. But there is one other place, the debugger, that calls this function directly. I'm going to do some more manual tests to confirm desired behavior.

Reply:
I was able to verify behavior, and this works as expected for code lenses and play button. Run menu is still not working, but that is a separate issue I'll address in follow up.